src/pkg/bundle/deploy.go

```diff
@@ -541,3 +485,200 @@ func formFilePath(anchorPath string, filePath string) (string, error) {
+	if err := goyaml.Unmarshal(bundleYAML, &b.bundle); err != nil {
+		return "", "", "", err
+	}
+
+	// validate bundle's arch against cluster
+	err = ValidateArch(config.GetArch(b.bundle.Build.Architecture))
+	if err != nil {
+		return "", "", "", err
+	}
+
+	bundleName := b.bundle.Metadata.Name
+	return bundleName, string(bundleYAML), source, err
+}
+
+// ConfirmBundleDeploy prompts the user to confirm bundle creation
+func (b *Bundle) ConfirmBundleDeploy() (confirm bool) {
+
+	pkgviews := formPkgViews(b)
+
+	message.HeaderInfof("🎁 BUNDLE DEFINITION")
+	pterm.Println("kind: UDS Bundle")
+
+	message.HorizontalRule()
+
+	message.Title("Metatdata:", "information about this bundle")
+	utils.ColorPrintYAML(b.bundle.Metadata, nil, false)
+
+	message.HorizontalRule()
+
+	message.Title("Build:", "info about the machine, UDS version, and the user that created this bundle")
+	utils.ColorPrintYAML(b.bundle.Build, nil, false)
+
+	message.HorizontalRule()
+
+	message.Title("Packages:", "definition of packages this bundle deploys, including variable overrides")
+
+	for _, pkg := range pkgviews {
+		utils.ColorPrintYAML(pkg.meta, nil, false)
+		utils.ColorPrintYAML(pkg.overrides, nil, false)
+	}
+
+	message.HorizontalRule()
+
+	// Display prompt if not auto-confirmed
+	if config.CommonOptions.Confirm {
+		return config.CommonOptions.Confirm
+	}
+
+	prompt := &survey.Confirm{
+		Message: "Deploy this bundle?",
+	}
+
+	if err := survey.AskOne(prompt, &confirm); err != nil || !confirm {
+		return false
+	}
+	return true
+}
+
+type PkgView struct {
+	meta      map[string]string
+	overrides map[string]interface{}
+}
+
+// formPkgViews creates a unique pre deploy view of each packages set overrides and Zarf variables
+func formPkgViews(b *Bundle) []PkgView {
+	var pkgViews []PkgView
+	for _, pkg := range b.bundle.Packages {
+		pkgMeta := map[string]string{"name": pkg.Name, "ref": pkg.Ref}
+		variables := make([]interface{}, 0)
+
+		if pkg.Repository != "" {
+			pkgMeta["repo"] = pkg.Repository
+		} else {
+			pkgMeta["path"] = pkg.Path
+		}
+
+		// process overrides and variables to get values
+		_, pkgOvers := b.loadVariables(pkg, nil)
+		valuesOverrides, _, _ := b.loadChartOverrides(pkg, map[string]string{})
+
+		// updated on each chart iteration
+		filteredVars := pkgOvers
+
+		for compName, component := range pkg.Overrides {
+			for chartName, chart := range component {
+				// filter out the chart.Variables so left with Zarf Variables
+				filteredVars = filterOverrides(chart.Variables, filteredVars)
+
+				processedVars := valuesOverrides[compName][chartName]
+				if processedVars == nil {
+					continue
+				}
+
+				chartVars := map[string]interface{}{}
+
+				for _, v := range chart.Variables {
+
+					// Mask potentially sensitive variables
+					if v.Type == "file" || v.Source == valuesources.Env {
+						chartVars[v.Name] = "****"
```

Comment:
I noticed that we set `v.Name` to `****`, but is the name actually sensitive or just the value?

Reply:
silly comment, disregard